packages/berry-pnpify/sources/NodePathResolver.ts

```diff
@@ -78,8 +81,32 @@ export class NodePathResolver {
     this.options = options;
   }
 
-  private getIssuer(pnp: PnpApi, pathname: string): string | undefined {
-    const locator = pnp.findPackageLocator(pathname + '/');
+  /**
+   * Returns `readdir`-like result for partially resolved pnp path
+   *
+   * @param issuerInfo issuer package information
+   * @param request either '' or '@scope'
+   *
+   * @returns `undefined` - if dir does not exist, or `readdir`-like list of subdirs in the virtual dir
+   */
+  public readDir(issuerInfo: PackageInformation, request: string): string[] | undefined {
```

Comment:
Using `request` as parameter name is a bit confusing since a request is potentially a long path (`@foo/bar/index.js`). I would suggest to call it `scope` instead.

Giving meaning to empty strings is also discouraged as it goes against the typing system. What do you think of making it a `string | null` parameter instead?

Reply:
Sure, sounds good!